### ISSUE TITLE
Several trial tests fail when run multiple times in a single process

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2481]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2481|
|Type|defect|
|Created|2007-02-25 17:52:08Z|
|Branch|https://github.com/twisted/twisted/tree/repeatable-loader-tests-2481|

```
===============================================================================
[FAIL]: twisted.trial.test.test_class.NumberOfRuns.test_1

Traceback (most recent call last):
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/test/test_class.py", line 14, in test_1
    self.failUnlessEqual(_setUpClassRuns, 1)
twisted.trial.unittest.FailTest: 2 != 1
===============================================================================
[FAIL]: twisted.trial.test.test_class.NumberOfRuns.test_2

Traceback (most recent call last):
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/test/test_class.py", line 18, in test_2
    self.failUnlessEqual(_setUpClassRuns, 1)
twisted.trial.unittest.FailTest: 2 != 1
===============================================================================
[FAIL]: twisted.trial.test.test_class.NumberOfRuns.test_3

Traceback (most recent call last):
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/test/test_class.py", line 22, in test_3
    self.failUnlessEqual(_setUpClassRuns, 1)
twisted.trial.unittest.FailTest: 2 != 1
===============================================================================
[FAIL]: twisted.trial.test.test_loader.FileTest.test_moduleNotInPath

Traceback (most recent call last):
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/test/test_loader.py", line 89, in test_moduleNotInPath
    os.path.splitext(sample1.__file__)[0])
twisted.trial.unittest.FailTest: 'twisted.trial.test.test_loader/FileTest/test_moduleNotInPath/kPXZdt/temp/goodpackage/test_sample' != 'twisted.trial.test.test_loader/FileTest/test_moduleNotInPath/Kc5m6o/temp/goodpackage/test_sample'
===============================================================================
[FAIL]: twisted.trial.test.test_loader.FileTest.test_packageNotInPath

Traceback (most recent call last):
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/test/test_loader.py", line 104, in test_packageNotInPath
    os.path.splitext(package1.__file__)[0])
twisted.trial.unittest.FailTest: 'twisted.trial.test.test_loader/FileTest/test_packageNotInPath/js42MK/temp/goodpackage/__init__' != 'twisted.trial.test.test_loader/FileTest/test_packageNotInPath/He1m4O/temp/goodpackage/__init__'
-------------------------------------------------------------------------------
```

<details><summary>Searchable metadata</summary>

```
trac-id__2481 2481
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_repeatable_loader_tests_2481 branches/repeatable-loader-tests-2481
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__tests tests
time__1172425928000000 1172425928000000
changetime__1261146662000000 1261146662000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

The `twisted.trial.test.test_class.NumberOfRuns` tests are all going to be removed, as they're for `setUpClass` and `tearDownClass` functionality which is being removed.  The `test_loader` failures, however, remain.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [27803]) Branching to 'repeatable-loader-tests-2481'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

"Fixed" the `test_loader` failures.  There are already some other tickets relating to `filenameToModule` (a ticket search for that name will produce quite a list), so I'm keeping the scope of this ticket to just making the tests pass when run twice.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Nice fix, +1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [27819]) Merge repeatable-loader-tests-2481

Author: exarkun
Reviewer: therve
Fixes: [#2481](https://github.com/twisted/twisted/issues/2481)

Clean up sys.modules in a couple trial tests so that the weird, semi-bogus state they
create doesn't influence other tests or those same tests if they're run again.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
